nyxtools/isara_robot.py

```diff
@@ -0,0 +1,313 @@
+import bluesky.plan_stubs as bps
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsSignal, EpicsSignalRO
+
+# TIMEOUT in seconds, should be declared elsewhere
+ISARA_TIMEOUT = 100
+
+
+class IsaraRobotDevice(Device):
+    # Commands
+
+    # Generic command channels
+    # power
+    power_on = Cpt(EpicsSignal, "Pwr:On-Cmd")
+    power_off = Cpt(EpicsSignal, "Pwr:Off-Cmd")
+
+    # arm movement speed
+    speed_up = Cpt(EpicsSignal, "Spd:Up-Cmd")
+    speed_down = Cpt(EpicsSignal, "Spd:Dn-Cmd")
+    # 0 - 100
+    speed_setpoint = Cpt(EpicsSignal, "Speed-SP")
+
+    # heater
+    heater_on = Cpt(EpicsSignal, "Htr:On-Cmd")
+    heater_off = Cpt(EpicsSignal, "Htr:Off-Cmd")
+
+    # dewar lid
+    dewar_lid_open = Cpt(EpicsSignal, "Lid:Opn-Cmd")
+    dewar_lid_close = Cpt(EpicsSignal, "Lid:Cls-Cmd")
+
+    # gripper a
+    gripper_a_open = Cpt(EpicsSignal, "OpnA-Cmd")
+    gripper_a_close = Cpt(EpicsSignal, "ClsA-Cmd")
+
+    # gripper b
+    gripper_b_open = Cpt(EpicsSignal, "OpnB-Cmd")
+    gripper_b_close = Cpt(EpicsSignal, "ClsB-Cmd")
+
+    # Trajectories
+    # write 1 to start move
+    home_traj = Cpt(EpicsSignal, "Move:Home-Cmd", put_complete=True)
+    recover_traj = Cpt(EpicsSignal, "Move:Rcvr-Cmd", put_complete=True)
+    get_traj = Cpt(EpicsSignal, "Move:Get-Cmd", put_complete=True)
+    put_traj = Cpt(EpicsSignal, "Move:Put-Cmd", put_complete=True)
+    getput_traj = Cpt(EpicsSignal, "Move:GetPut-Cmd", put_complete=True)
+    back_traj = Cpt(EpicsSignal, "Move:Bck-Cmd", put_complete=True)
+    dry_traj = Cpt(EpicsSignal, "Move:Dry-Cmd", put_complete=True)
+    soak_traj = Cpt(EpicsSignal, "Move:Sk-Cmd", put_complete=True)
+    pick_traj = Cpt(EpicsSignal, "Move:Pck-Cmd", put_complete=True)
+
+    # Trajectory Arguments
+    # 0 = "ToolChanger"
+    # 1 = "Cryotong"
+    # 2 = "SingleGripper"
+    # 3 = "DoubleGripper"
+    # 4 = "MiniSpineGripper"
+    # 5 = "RotatingGripper"
+    # 6 = "PlateGripper"
+    # 7 = "Spare"
+    # 8 = "LaserTool"
+    tool_selected = Cpt(EpicsSignal, "Tl-Sel")
+
+    # limits 0-29
+    # argument is interchangeable for puck/plate selection field
+    puck_num_sel = Cpt(EpicsSignal, "Plt-SP", put_complete=True)
+    puck_n_selected = Cpt(EpicsSignal, "Plt:N-SP", put_complete=True)
```

Comment:
what is the difference between these two puck_x_sel PVs?

feel free to add a manual to the code base and/or use text from there

Reply:
in hindsight this is the worst possible set of names, I will fix that

Reply:
fixed with 6b200ff